tests/testthat/test-climatic_summary.R

```diff
@@ -1,43 +1,94 @@
-y_daily <- climatic_summary(data=daily_niger,date_time ="date", by = c("station_name"), station = "station_name", to = "daily", elements ="rain", summaries = c(mean = "mean", st_dv = "sd", n_na = "naflex::na_n"))
-y_pentad <- climatic_summary(data=daily_niger,date_time ="date", by = c("station_name"), station = "station_name", to = "pentad", elements ="rain", summaries = c(mean = "mean", st_dv = "sd", n_na = "naflex::na_n"))
-y_dekadal <- climatic_summary(data=daily_niger,date_time ="date", by = c("station_name"), station = "station_name", to = "dekadal", elements ="rain", summaries = c(mean = "mean", st_dv = "sd", n_na = "naflex::na_n"))
-y_monthly <- climatic_summary(data=daily_niger,date_time ="date", by = c("station_name"), station = "station_name", to = "monthly", elements ="rain", summaries = c(mean = "mean", st_dv = "sd", n_na = "naflex::na_n"))
-y_annual_within_year <- climatic_summary(data=daily_niger,date_time ="date", by = c("station_name"), station = "station_name", to = "annual-within-year", elements ="rain", summaries = c(mean = "mean", st_dv = "sd", n_na = "naflex::na_n"))
-y_annual <- climatic_summary(data=daily_niger,date_time ="date", by = c("station_name"), station = "station_name", to = "annual", elements ="rain", summaries = c(mean = "mean", st_dv = "sd", n_na = "naflex::na_n"))
-y_longterm_monthly <- climatic_summary(data=daily_niger,date_time ="date", by = c("station_name"), station = "station_name", to = "longterm-monthly", elements ="rain", summaries = c(mean = "mean", st_dv = "sd", n_na = "naflex::na_n"))
-y_longterm_within_year <- climatic_summary(data=daily_niger,date_time ="date", by = c("station_name"), station = "station_name", to = "longterm-within-year", elements ="rain", summaries = c(mean = "mean", st_dv = "sd", n_na = "naflex::na_n"))
-y_station <- climatic_summary(data=daily_niger,date_time ="date", by = c("station_name"), station = "station_name", to = "station", elements ="rain", summaries = c(mean = "mean", st_dv = "sd", n_na = "naflex::na_n"))
-y_overall <- climatic_summary(data=daily_niger,date_time ="date", by = c("station_name"), station = "station_name", to = "overall", elements ="rain", summaries = c(mean = "mean", st_dv = "sd", n_na = "naflex::na_n"))
+x_daily <- niger %>% 
+  group_by(date) %>% 
+  summarise(mean_rain = mean(rain), st_dv_rain = sd(rain), n_na_rain = naflex::na_n(rain))
+
+x_monthly <- niger %>% 
+  group_by(station_name, year, month) %>% 
+  summarise(mean_rain = mean(rain), st_dv_rain = sd(rain), n_na_rain = naflex::na_n(rain))
+
+x_annual <- niger %>% 
+  group_by(station_name, year) %>% 
+  summarise(mean_rain = mean(rain), st_dv_rain = sd(rain), n_na_rain = naflex::na_n(rain))
+
+x_station <- niger %>% 
+  group_by(station_name) %>% 
+  summarise(mean_rain = mean(rain), st_dv_rain = sd(rain), n_na_rain = naflex::na_n(rain))
```

Comment:
@HawardKetoyoMsatsi See here how I fixed the tests. For many examples you weren't grouping by station too.